client/packages/electron/src/electron.ts

```diff
@@ -184,22 +188,39 @@ const start = (): void => {
   );
 
   ipcMain.handle(IPC_MESSAGES.CONNECTED_SERVER, async () => connectedServer);
-  ipcMain.handle(IPC_MESSAGES.START_BARCODE_SCAN, () =>
-    barcodeScanner.start(window)
-  );
-  ipcMain.handle(IPC_MESSAGES.STOP_BARCODE_SCAN, () => barcodeScanner.stop());
 
   ipcMain.handle(IPC_MESSAGES.DISCOVERED_SERVERS, async () => {
     const servers = discoveredServers;
     discoveredServers = [];
     return { servers };
   });
 
+  // BARCODES
+  const serialScanner = new Scanner(window);
+  const keyboardScanner = new KeyboardScanner(window);
+  const getCurrentScanner = () =>
+    store.get(SCANNER_TYPE, 'usb_serial') == 'usb_serial'
+      ? serialScanner
+      : keyboardScanner;
+
+  ipcMain.on(
+    IPC_MESSAGES.SET_SCANNER_TYPE,
+    (_event, scannerType: ScannerType) => store.set(SCANNER_TYPE, scannerType)
+  );
   ipcMain.handle(IPC_MESSAGES.LINKED_BARCODE_SCANNER_DEVICE, async () =>
-    barcodeScanner.linkedScanner()
+    getCurrentScanner().linkedScanner()
+  );
+  ipcMain.handle(IPC_MESSAGES.START_BARCODE_SCAN, () =>
+    getCurrentScanner().start()
+  );
+  ipcMain.handle(IPC_MESSAGES.STOP_BARCODE_SCAN, () =>
+    getCurrentScanner().stop()
```

Comment:
These, and GET_SCANNER_TYPE & SET_SCANNER_TYPE, could all be handled with one IPC message if
- the LINKED_BARCODE_SCANNER_DEVICE was replaced with a call to get BARCODE_SCANNER which called `getCurrentScanner` 
- and the methods on that class were called by the caller (BarcodeScannerContext in this case)
- the class handled the scanner type: added to the constructor and methods added to get/set